MP Health test development

istraka opened the pull request:
refactoring
	negative scenario for MIME types
	multiple deployment / deployment with subdeployments scenario
	new health version test
	JavaDoc

https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-simple-face/11/

Please make sure your PR meets the following requirements:
- [x] Pull Request contains description for the changes
- [x] Pull Request does not include fixes for multiple issues / topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

istraka commented:
Jenkins is down for now. Here is a distribution (available only from RedHat network) http://download.eng.brq.redhat.com/scratch/istraka/wildfly-mp-ft-health.zip. 

The distribution is a merge of https://github.com/wildfly/wildfly/pull/12800 and https://github.com/wildfly/wildfly/pull/12720

fabiobrz commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SimplifiedHealthTest.java:
```diff
@@ -0,0 +1,96 @@
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(2), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```
`BothHealthCheck` -> `SimplifiedHealthTest` here?

istraka replied:
Fixing...

fabiobrz commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthTest.java:
```diff
@@ -39,28 +39,37 @@ public void composeHealthEndpointURL() {
     }
 
+    /**
+     * @tpTestDetails Customer scenario where health check contains customer data.
```
`health check` -> `liveness check`?

istraka replied:
Fixing...

fabiobrz commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthTest.java:
```diff
@@ -74,6 +83,11 @@ public void testLivenessEndpoint() {
     }
 
+    /**
+     * @tpTestDetails Customer scenario where health check contains customer data.
```
`health check` -> `readiness check`?

istraka replied:
Fixing...

istraka commented on microprofile-health/pom.xml:
```diff
@@ -39,5 +39,18 @@
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.extras.creaper</groupId>
+            <artifactId>creaper-commands</artifactId>
```
Yes, I have already developed tests that needs this. They will be part of following PR.

fabiobrz replied:
Ok, thanks. So I'll leave the last word regarding "just creaper-core" vs "creaper-core + creaper-commands" approach to @mnovak1 and @rsvoboda who brought up the topic earlier.

mnovak1 replied:
@istraka As decided on https://github.com/jboss-eap-qe/eap-microprofile-test-suite/issues/26 we should use just Creaper core. 

If Creaper commands are needed as well then we need strong arguments for it.

rsvoboda replied:
+1 for Mirek

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthNullTest.java:
```diff
@@ -21,21 +21,26 @@
     @Deployment(testable = false)
     public static Archive<?> deployment() {
-        WebArchive war = ShrinkWrap.create(WebArchive.class, "MicroProfileHealthNullTest.war")
+        WebArchive war = ShrinkWrap.create(WebArchive.class, HealthTest.class.getSimpleName() + ".war")
```
Yes, thanks!

istraka replied:
Fixing...

istraka commented on pom.xml:
```diff
@@ -122,6 +123,16 @@
+            </dependency>
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-io</artifactId>
```
Tests brought by next iteration will use it.

fabiobrz replied:
Ok, thanks.

rsvoboda replied:
> Tests brought by next iteration will use it.

please don't add it here, viz "Pull Request does not include fixes for multiple issues / topics" checkbox

istraka commented on pom.xml:
```diff
@@ -132,6 +143,9 @@
                 <configuration>
+                    <environmentVariables>
+                        <JBOSS_HOME>${jboss.home}</JBOSS_HOME>
+                    </environmentVariables>
```
Because of Creaper. Not used now, but in following PR it will be needed.

fabiobrz replied:
Can you add a short explanation for this? I think it is needed for it to be shared here. Thanks.

istraka replied:
I can not execute CLI command using Creaper
```
ERROR: Setup failed during setup. Offending class 'org.jboss.eap.qe.microprofile.health.integration.FailSafeCDICustomConfigSourceHealthTest$SetupTask@381d7219'
org.wildfly.extras.creaper.core.CommandFailedException: org.wildfly.extras.creaper.core.online.CliException: org.jboss.as.cli.CommandLineException: java.util.concurrent.ExecutionException: org.jboss.as.cli.CommandLineException: JBOSS_HOME environment variable is not set.
	at org.wildfly.extras.creaper.core.online.AutomaticErrorHandlingForCommands.commandFailedWithCause(AutomaticErrorHandlingForCommands.java:142)
	at org.wildfly.extras.creaper.core.online.AutomaticErrorHandlingForCommands.executeCli(AutomaticErrorHandlingForCommands.java:109)
	at org.wildfly.extras.creaper.commands.modules.AddModule.apply(AddModule.java:91)
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.apply(OnlineManagementClientImpl.java:136)
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.apply(OnlineManagementClientImpl.java:125)
	at org.wildfly.extras.creaper.core.online.LazyOnlineManagementClient.apply(LazyOnlineManagementClient.java:50)
	at org.jboss.eap.qe.microprofile.health.integration.ModuleUtil.setupModule(ModuleUtil.java:21)
	at org.jboss.eap.qe.microprofile.health.integration.FailSafeCDICustomConfigSourceHealthTest$SetupTask.setup(FailSafeCDICustomConfigSourceHealthTest.java:71)
	at org.jboss.as.arquillian.container.ServerSetupObserver$ServerSetupTaskHolder.setup(ServerSetupObserver.java:185)
	at org.jboss.as.arquillian.container.ServerSetupObserver.handleBeforeDeployment(ServerSetupObserver.java:108)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:147)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$3.call(ContainerDeployController.java:118)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.executeOperation(ContainerDeployController.java:239)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deploy(ContainerDeployController.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createDeploymentContext(ContainerDeploymentContextHandler.java:71)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.container.impl.client.container.DeploymentExceptionHandler.verifyExpectedExceptionDuringDeploy(DeploymentExceptionHandler.java:47)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:92)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController$1.perform(ContainerDeployController.java:77)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachDeployment(ContainerDeployController.java:232)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.forEachManagedDeployment(ContainerDeployController.java:212)
	at org.jboss.arquillian.container.impl.client.container.ContainerDeployController.deployManaged(ContainerDeployController.java:77)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:96)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:89)
	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:163)
	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: org.wildfly.extras.creaper.core.online.CliException: org.jboss.as.cli.CommandLineException: java.util.concurrent.ExecutionException: org.jboss.as.cli.CommandLineException: JBOSS_HOME environment variable is not set.
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.executeCli(OnlineManagementClientImpl.java:210)
	at org.wildfly.extras.creaper.core.online.AutomaticErrorHandlingForCommands.executeCli(AutomaticErrorHandlingForCommands.java:107)
	... 107 more
Caused by: org.jboss.as.cli.CommandLineException: java.util.concurrent.ExecutionException: org.jboss.as.cli.CommandLineException: JBOSS_HOME environment variable is not set.
	at org.jboss.as.cli.impl.CommandContextImpl.execute(CommandContextImpl.java:878)
	at org.jboss.as.cli.impl.CommandContextImpl.handleLegacyCommand(CommandContextImpl.java:1840)
	at org.jboss.as.cli.impl.CommandContextImpl.handle(CommandContextImpl.java:815)
	at org.jboss.as.cli.impl.WorkaroundForWFCORE526_CommandContextImpl.handle(WorkaroundForWFCORE526_CommandContextImpl.java:14)
	at org.wildfly.extras.creaper.core.online.OnlineManagementClientImpl.executeCli(OnlineManagementClientImpl.java:203)
	... 108 more
Caused by: java.util.concurrent.ExecutionException: org.jboss.as.cli.CommandLineException: JBOSS_HOME environment variable is not set.
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.jboss.as.cli.impl.CommandExecutor.execute(CommandExecutor.java:713)
	at org.jboss.as.cli.impl.CommandExecutor.execute(CommandExecutor.java:694)
	at org.jboss.as.cli.impl.CommandContextImpl.lambda$handleLegacyCommand$4(CommandContextImpl.java:1841)
	at org.jboss.as.cli.impl.CommandContextImpl.execute(CommandContextImpl.java:865)
	... 112 more
Caused by: org.jboss.as.cli.CommandLineException: JBOSS_HOME environment variable is not set.
	at org.jboss.as.cli.handlers.module.ASModuleHandler.getModulesDir(ASModuleHandler.java:570)
	at org.jboss.as.cli.handlers.module.ASModuleHandler.addModule(ASModuleHandler.java:395)
	at org.jboss.as.cli.handlers.module.ASModuleHandler.doHandle(ASModuleHandler.java:346)
	at org.jboss.as.cli.handlers.CommandHandlerWithHelp.handle(CommandHandlerWithHelp.java:89)
	at org.jboss.as.cli.impl.CommandExecutor$2.lambda$build$0(CommandExecutor.java:685)
	at org.jboss.as.cli.impl.CommandExecutor.lambda$execute$0(CommandExecutor.java:708)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

fabiobrz replied:
Thanks, not really "short" but definitely valuable :)

istraka commented on pom.xml:
```diff
@@ -122,6 +123,16 @@
             </dependency>
+            <dependency>
+                <groupId>org.wildfly.extras.creaper</groupId>
+                <artifactId>creaper-commands</artifactId>
```
Next iteration.

fabiobrz replied:
OK thanks, same as above.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthTest.java:
```diff
@@ -39,28 +39,37 @@ public void composeHealthEndpointURL() {
+    /**
+     * @tpTestDetails Customer scenario where health check contains customer data.
+     * @tpPassCrit Overall and the health check status is up and checks contains expected data.
+     * @tpSince EAP 7.3.0.CD19
```
Fixing...

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiAppHealthTest.java:
```diff
@@ -0,0 +1,82 @@
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(3), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```
Yes, I see your point. The abstract class tests expect sub-classes to deploy something specific / limited. What about having a sort of TestDelegation class instead of this abstract class?

fabiobrz replied:
> The abstract class tests expect sub-classes to deploy something specific / limited.

Correct. The key point of my consideration is to not scatter implementation logic over hierarchy. 
Any "pattern" that would solve this is good IMO. Delegation to run some supplied code would work, I guess, but then what's the point for that "empty" superclass? 
I - for instance - would start from removing the "details-aware" code from the superclass to subclasses - i.e. I'd move the checks down, but that would just be my immediate coding attempt. 
Then the design could be adjusted further, somebody else could choose composition. 
I'd say there are several solutions and you know better the detail.

istraka replied:
Yeah, I would get rid of the hierarchy....Fixing...

istraka commented:
@fabiobrz I have addressed issues. At the end I haven't implemented any sort of test delegation class since it is too complicated for the purpose. I just get rid of the hierarchy,

fabiobrz commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiDeploymentHealthTest.java:
```diff
@@ -0,0 +1,106 @@
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(3), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```
Here again I think it's `BothHealthCheck` -> `SimplifiedHealthCheck` or just remove the comment.

istraka replied:
Fixing...

fabiobrz commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SubdeploymentHealthTest.java:
```diff
@@ -0,0 +1,125 @@
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(3), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```
BothHealthCheck -> SimplifiedHealthCheck or just remove the comment.

istraka replied:
Fixing...

istraka commented:
Comments were removed.

rsvoboda commented:
@istraka rebase please

mnovak1 commented:
@istraka I would like to see improved JavaDoc with test description so it's clear what the test is checking. Most of the javadoc is too general.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthNullTest.java:
```diff
@@ -21,21 +21,26 @@
+public class HealthNullTest {
 
     @ContainerResource
     ManagementClient managementClient;
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SimplifiedHealthTest.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    @Test
+    public void testHealthEndpoint() {
+        RestAssured.when().get(healthURL).then()
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthNullTest.java:
```diff
@@ -21,21 +21,26 @@
     @Test
-    public void testDeprecatedHealthCheck() {
+    public void testNullHealthCheck() {
         final String healthURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort()
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MimeTypeHealthTest.java:
```diff
@@ -0,0 +1,163 @@
+    }
+
+    /**
+     * @tpTestDetails Negative scenario where health check contains customer data. Request accepts invalid MIME type
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MimeTypeHealthTest.java:
```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * @tpTestDetails Negative scenario where health check contains customer data. Request accepts invalid MIME type
+     * @tpPassCrit Overall and the health check status is up and checks contains expected data.
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiDeploymentHealthTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * @tpTestDetails Customer multi-deployment scenario.
```
Fixed.

mnovak1 replied:
Hmm...could you provide more info what endpoint the test is checking like:
```
Customer multi-deployment scenario. Checks health endpopint from more deployments are registered and exposed.
```

istraka replied:
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiDeploymentHealthTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * @tpTestDetails Customer multi-deployment scenario.
```
Fixed.

istraka commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiDeploymentHealthTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    /**
+     * @tpTestDetails Customer multi-deployment scenario.
```
Fixed.

istraka commented:
@mnovak1  I have addressed issues. Please review.

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthDeprecatedTest.java:
```diff
@@ -22,21 +23,26 @@
     }
 
+    /**
+     * @tpTestDetails backward compatibility scenario - test for deprecated however still supported annotation.
```
Could you mention here which annotation is deprecated, please?

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthDeprecatedTest.java:
```diff
@@ -22,21 +23,26 @@
+     */
     @Test
-    public void testDeprecatedHealthCheck() {
+    public void testDeprecatedHealthCheck() throws ConfigurationException {
```
Maybe `testDeprecatedHealthCheck` -> `testDeprecatedHealthAnnotation`

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthUtil.java:
```diff
@@ -0,0 +1,34 @@
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+
+public class HealthUtil {
```
`HealthUtil` sounds quite generic, could you rename to something like `HealthUrlProvider`, please?

istraka replied:
yep

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthUtil.java:
```diff
@@ -0,0 +1,34 @@
+        return new URLProvider();
+    }
+
+    public static class URLProvider {
```
Is it possible to get rid of nested static class and rather make all methods static so in test this would be used like:
```
HealthUrlProvider.getHealthEndpoint()
```

istraka replied:
Sure.

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SimplifiedHealthTest.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Deployment(testable = false)
+    public static Archive<?> deployment() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, SimplifiedHealthTest.class + ".war")
```
Optional - inline war. There is no need for creating war and it can be returned `ShrinkWrap.create(WebArchive.cl...`

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthDeprecatedTest.java:
```diff
@@ -22,21 +23,26 @@
     @Test
-    public void testDeprecatedHealthCheck() {
+    public void testDeprecatedHealthCheck() throws ConfigurationException {
         final String healthURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort()
```
I guess this can be removed now.

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/HealthUrlProvider.java:
```diff
@@ -0,0 +1,27 @@
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+
+public class HealthUrlProvider {
```
Could you move it to subpackage, something like `tooling`|`helper`, please?

istraka replied:
Done

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/SimplifiedHealthTest.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * @tpTestDetails Smoke test for new api.
```
This is the last javadoc to improve.

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/tools/HealthUrlProvider.java:
```diff
@@ -0,0 +1,27 @@
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+
+public class HealthUrlProvider {
```
And some javadoc should be here.

istraka commented:
Fixes !14
